packages/yourloops/layout/hcp-layout-with-context.tsx

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2023, Diabeloop
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React, { type FunctionComponent } from 'react'
+import { SelectedTeamProvider } from '../lib/selected-team/selected-team.provider'
+import { HcpLayout } from './hcp-layout'
+import { TeamContextProvider } from '../lib/team'
+
+export const HcpLayoutWithContext: FunctionComponent = () => {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (non-blocking):** ‏What I would have done instead of creating a new component with a "blurry" naming is:

```
const getUserLayout = (): JSX.Element => {
    switch (user?.role) {
      case UserRole.Hcp:
        return <TeamContextProvider>
          <SelectedTeamProvider>
            <HcpLayout />
          </SelectedTeamProvider>
        </TeamContextProvider>
      case UserRole.Caregiver:
        return <CaregiverLayout />
      case UserRole.Patient:
        return <PatientLayout />
      default:
        console.error(`no layout found for role ${user?.role}`)
        return <Route
          path="*"
          element={<Navigate to="/not-found" replace />}
        />
    }
  }
```

In `packages/yourloops/layout/main-layout.tsx`.
But your solution is also fine :wink: